### ISSUE TITLE
Quest Bug when handing in some Pet Battle Quests, though not always - issue since WoW Patch 8.1

### ISSUE THREAD

**Issue by Swirrel** [mikepauer/Carbonite.Quests#58]
5x Carbonite.Quests\NxQuest.lua:11889: attempt to compare number with nil
Carbonite.Quests\NxQuest.lua:11889: in function `Update'
Carbonite.Quests\NxQuest.lua:11858: in function <Carbonite.Quests\NxQuest.lua:11858>

Locals:
self = <table> {
 UpdateDB = <function> defined @Carbonite.Quests\NxQuest.lua:11816
 Win = <table> {
 }
 RegisterMessage = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 UnregisterMessage = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 Update = <function> defined @Carbonite.Quests\NxQuest.lua:11872
 List = <table> {
 }
 SendMessage = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:59
 ButMenu = <table> {
 }
 GenWQTip = <function> defined @Carbonite.Quests\NxQuest.lua:11673
 UnregisterAllEvents = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 RegisterEvent = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 UnregisterAllMessages = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 Open = <function> defined @Carbonite.Quests\NxQuest.lua:11566
 UnregisterEvent = <function> defined @AtlasLoot\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 OnListEvent = <function> defined @Carbonite.Quests\NxQuest.lua:11806
 FlashColor = 0
 Menu = <table> {
 }
 FirstUpdate = true
 WinUpdateFade = <function> defined @Carbonite.Quests\NxQuest.lua:11667
 CheckBounty = <function> defined @Carbonite.Quests\NxQuest.lua:11861
 GetWQReward = <function> defined @Carbonite.Quests\NxQuest.lua:11767
 Opened = true
}
list = <table> {
 SelFrm = <unnamed> {
 }
 LineHPad = 0
 MinW = 200
 Vis = 144
 HdrH = 0
 Data = <table> {
 }
 BaseLineH = 12
 UserFunc = <function> defined @Carbonite.Quests\NxQuest.lua:11806
 ShowAll = true
 Selected = 0
 Num = 1
 Columns = <table> {
 }
 User = <table> {
 }
 SSW = 672.999984
 ButData = <table> {
 }
 Buts = <table> {
 }
 UsedFrms = <table> {
 }
 Top = 1
 Font = "QuestWatch.WatchFont"
 SSH = 1872
 Frm = <unnamed> {
 }
 FontObj = NxFontW {
 }
 Sorted = false
 MinH = 20
 Strs = <table> {
 }
}
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 51637 = <table> {
 }
 50633 = <table> {
 }
 44011 = <table> {
 }
 48099 = <table> {
 }
 51175 = <table> {
 }
 44816 = <table> {
 }
 51972 = <table> {
 }
 43063 = <table> {
 }
 54546 = <table> {
 }
 41549 = <table> {
 }
 54594 = <table> {
 }
 51558 = <table> {
 }
 48514 = <table> {
 }
 51096 = <table> {
 }
 43454 = <table> {
 }
 44737 = <table> {
 }
 48833 = <table> {
 }
 43247 = <table> {
 }
 45550 = <table> {
 }
 44801 = <table> {
 }
 42028 = <table> {
 }
 48467 = <table> {
 }
 46180 = <table> {
 }
 42108 = <table> {
 }
 54619 = <table> {
 }
 49041 = <table> {
 }
 52874 = <table> {
 }
 51615 = <table> {
 }
 50619 = <table> {
 }
 41965 = <table> {
 }
 48101 = <table> {
 }
 43798 = <table> {
 }
 41766 = <table> {
 }
 43583 = <table> {
 }
 52779 = <table> {
 }
 52803 = <table> {
 }
 42101 = <table> {
 }
 44157 = <table> {
 }
 52325 = <table> {
 }
 47496 = <table> {
 }
 51584 = <table> {
 }
 41145 = <table> {
 }
 51632 = <table> {
 }
 41950 = <table> {
 }
 51664 = <table> {
 }
 50660 = <table> {
 }
 44301 = <table> {
 }
 52756 = <table> {
 }
 42819 = <table> {
 }
 52007 = <table> {
 }
 43600 = <table> {
 }
 51776 = <table> {
 }
 51529 = <table> {
 }
 47712 = <table> {
 }
 51832 = <table> {
 }
 43943 = <table> {
 }
 42182 = <table> {
 }
 44748 = <table> {
 }
 45792 = <table> {
 }
 46063 = <table> {
 }
 51434 = <table> {
 }
 42015 = <table> {
 }
 42788 = <table> {
 }
 52757 = <table> {
 }
 42820 = <table> {
 }
 53283 = <table> {
 }
 43346 = <table> {
 }
 50502 = <table> {
 }
 46175 = <table> {
 }
 43641 = <table> {
 }
 54614 = <table> {
 }
 48287 = <table> {
 }
 50853 = <table> {
 }
 50861 = <table> {
 }
 48829 = <table> {
 }
 42725 = <table> {
 }
 40980 = <table> {
 }
 52463 = <table> {
 }
 45379 = <table> {
 }
 46933 = <table> {
 }
 48734 = <table> {
 }
 52352 = <table> {
 }
 41658 = <table> {
 

I can see by skimming through the bug, that it also mentions AtlasLoot. I do have AtlasLoot though the addon in game doesn't give it's version number (sorry)
Here's a screenie of the AtlasLoot and it's sub-addons that I have turned on (normaly I have all the AtlasLoot sub-addons turned on, but I turned most of thursday evening and just haven't turned them on again.
![image](https://user-images.githubusercontent.com/31900265/50368079-8152fe80-0585-11e9-8616-67c841702e64.png)

**Comment by Thrumbar** [comment]
Related to this
#57 
And fixed in this
00a04505c249f886c58c308af7191fa956456fe4

**Comment by Swirrel** [comment]
With all due respect... that bug happend to me 3 days ago and the fix was applied 4 days ago.. which means that the fix was applied 1 day before my post.
It may or may not be the same type of bug or similar in type, but I wouldn't really get the same bug if the bug was fixed...
And yes, I got the newest Alpha for both Carbonite Maps and Carbonite Quest as those are the only 2 Carbonite addons I use.

This isn't to be annoying, merely to state the obvious.
